src/Controller/EntityCreateController.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity\Controller\EntityCreateController.
+ */
+
+namespace Drupal\entity\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Url;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * A generic controller for creating entities.
+ */
+class EntityCreateController extends ControllerBase {
+
+  /**
+   * Displays add links for the available bundles.
+   *
+   * Redirects to the add form if there's only one bundle available.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID.
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request.
+   *
+   * @return \Symfony\Component\HttpFoundation\RedirectResponse|array
+   *   If there's only one available bundle, a redirect response.
+   *   Otherwise, a render array with the add links for each bundle.
+   */
+  public function addPage($entity_type_id, Request $request) {
+    $entity_type = $this->entityTypeManager()->getDefinition($entity_type_id);
+    $bundle_type = $entity_type->getBundleEntityType();
+    $form_route_name = 'entity.' . $bundle_type . '.add_form';
+    $bundles = $this->entityTypeManager()->getStorage($bundle_type)->loadMultiple();
```

Comment:
We need to use \Drupal\Core\Entity\EntityTypeBundleInfoInterface instead of loading the bundles directly.